server/models/hackathonSchema.model.js

```diff
@@ -0,0 +1,91 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+/**
+ * @swagger
+ *  components:
+ *    schemas:
+ *      Hackathon:
+ *        type: object
+ *        properties:
+ *          season:
+ *            type: string
+ *            description: The season of the hackathon. (ex. Spring 2020, Fall 2019)
+ *          categories:
+ *            type: array
+ *            description: A list of categories in the hackathon
+ *            items:
+ *              type: object
+ *              properties:
+ *                name:
+ *                  type: String
+ *                  description: Name of the category.
+ *                companyName:
+ *                  type: String
+ *                  description: Name, if any, of the company that is sponsoring this category.
+ *                type:
+ *                  type: String
+ *                  description: Type of the category, can either be a track or a superlative.
+ *          submissionPhase:
+ *            type: object
+ *            description: Object that indicates whether the hackathon is in the
+ *                         submission phase, stores the deadline and stores the
+ *                         submissions that have been submitted.
+ *            properties:
+ *              inProgress:
+ *                type: boolean
+ *                default: false
+ *                description: Indicates if the submissions phase is in progress.
+ *              deadline:
+ *                type: Date
+ *                description: Deadline to submit submissions. If the deadline is
+ *                             reached, set inProgress to false.
+ *              flags:
+ *                type: array
+ *                description: Array of submissions that have been submitted.
+ *                items:
+ *                  type: ObjectId
+ *                  ref: submission
+ *          judgingPhase:
+ *            type: object
+ *            description: Object that indicates whether the judging phase is
+ *                         in progress, and also stores relevant data.
+ *            properties:
+ *              inProgress:
+ *                type: boolean
+ *                default: false
+ *                description: Indicates whether is judging phase is in progress.
+ *                             if submissionPhase is in progress, then this MUST
+ *                             be true.
+ *          isComplete:
+ *            type: boolean
+ *            default: false
+ *            description: Indicates whether the hackathon is over.
+ *
+ */
+const hackathonSchema = new Schema({
+  season: String, // ex. Spring 2021
+  categories: {
+    type: [
+      {
+        name: String,
+        companyName: String,
+        type: String,
+      },
+    ],
+    ref: 'category',
+  }, // _id refers to a Category.
```

Comment:
No need for a ref.

You can do one of two things:
1. (preferred) Remove the categoryschema and directly put it in here without the ref.
2. You can import the category schema, and use it in here. Like:
```
line 4: import CategorySchema from ./categorySchema.model.js
line 68: categories: [CategorySchema]
```
At least, I think you can do that. But I would just do what I said in 1. and delete the categoryschema entirely and put it in the hackathon schema (so just delete the ref).